lib/rules/no-unused-vars.js

```diff
@@ -126,6 +131,30 @@ module.exports = {
         }
 
         /**
+         * Determines if a variable has a sibling rest property
+         * @param {Variable} variable - EScope variable object.
+         * @returns {boolean} True if the variable is exported, false if not.
+         * @private
+         */
+        function hasRestSpreadSibling(variable) {
+            if (config.ignoreRestSiblings) {
+                const restProperties = new Set(["ExperimentalRestProperty", "RestProperty"]);
+
+                return variable.defs
+                    .filter(def => def.name.type === "Identifier")
+                    .some(def => (
+                        def.node.id &&
+                        def.node.id.type === "ObjectPattern" &&
+                        def.node.id.properties.length &&
+                        restProperties.has(def.node.id.properties[def.node.id.properties.length - 1].type) &&  // last property is a rest property
+                        def.node.id.properties.some(prop => prop.type === "Property" && prop.key.name === variable.name)  // variable is sibling of the rest property
```

Comment:
Oh I see, interesting point. I still think it would be better to check the parent type though -- renaming the variable might be necessary if a different variable is already defined with the given name. For example, in the code below, the variable has to be renamed to avoid shadowing the existing `window` variable:

```js
var { window: _, ...otherProps } = unrelatedObj;

// later...

window.addEventListener(/* ... */);
```

Also, in some cases the property name is computed, so a `name` check won't work:

```js
function omit(obj, key) {
  const { [key]: _, ...remainingProps } = obj;
  return remainingProps;
}
```